Mathlib/Algebra/Category/Ring/Center.lean

```diff
@@ -0,0 +1,135 @@
+/-
+Copyright (c) 2025 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import Mathlib.CategoryTheory.Adjunction.Limits
+import Mathlib.Algebra.Category.ModuleCat.Limits
+
+/-!
+# A categorical description of the center of a ring
+
+In this file we prove that the center of a ring `R` is isomorphic to `End (𝟭 R-Mod)` the
+endomorphism ring of the identity functor on the category of `R`-modules. Consequently, the ring
+structure of a commutative ring is complete determined by its module category.
+
+## Main results
+
+- `Subring.centerEquivEndIdFunctor`: the center of a ring `R` is isomorphic to `End (𝟭 R-Mod)`.
+- `RingEquiv.ofModuleCatEquiv`: if two commutative rings have equivalent module categories, they are
+  isomorphic as rings.
+
+-/
+
+universe u u' v v'
+
+variable (R : Type u) [Ring R]
+
+open CategoryTheory
+
+/--
+For any ring `R`, the center of `R` is isomorphic to `End (𝟭 (ModuleCat R))`, the endomorphism ring
+of the identity functor on the category of `R`-modules.
+-/
+@[simps]
+def Subring.centerEquivEndIdFunctor : Subring.center R ≃+* End (𝟭 (ModuleCat.{max v u} R)) where
+  toFun x :=
+    { app M := ModuleCat.ofHom
+        { toFun := (x.1 • ·)
+          map_add' := by aesop
+          map_smul' r m := by simp [← mul_smul, Subring.mem_center_iff.1 x.2 r] } }
+  map_add' x y := by
+    apply NatTrans.ext
+    ext M (m : M)
+    exact add_smul x.1 y.1 m
+  map_mul' x y := by
+    apply NatTrans.ext
+    ext M (m : M)
+    exact mul_smul x.1 y.1 m
+  invFun f := ⟨f.app (.of R <| ULift R) |>.hom (1 : ULift R) |>.down, by
```

Comment:
Can you use [Module.moduleEndSelf](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/LinearMap/End.html#Module.moduleEndSelf)? Then this is basically Z(Module.End R R) = Z(R^mop) = Z(R).

Reply:
I am not sure I follow. Is the idea that constructing a ring isomorphism between `Z(Module.End R R)` and `CategoryTheory.End (𝟭 (ModuleCat R))` is easier?

Reply:
I've worked it out in [this branch](https://github.com/leanprover-community/mathlib4/commit/5971b5d998673c2fb656cf6ecf0e6a3590e871c7)